### PR TITLE
Bugs and Improvements

### PR DESCRIPTION
**Pull request by CPKreuz**
Changes:
* Fixed .pixi preview quality (In the exporter)
* Automatically select newly created layer
* Fixed opacity slider not updating when clicking
* Added a command for pasting a reference layer from the clipboard
* Fixed documents not being at the start of the recently opened document list if they were already in the list
* Added a debug command for clearing the list of recently opened documents (Found in Debug > Delete)
* Added a window for debugging commands
* Added icons for the rotate commands
* Added some documentation

### CODE REVIEW

File: src/PixiEditor/ViewModels/SubViewModels/Main/LayersViewModel.cs [hunk 1]
```diff
@@ -347,6 +377,26 @@ public void ImportReferenceLayer()
...
     }
+
+    [Command.Basic("PixiEditor.Layer.PasteReferenceLayer", "Paste reference layer", "Paste reference layer from clipboard", IconPath = "Commands/PixiEditor/Clipboard/Paste.png", CanExecute = "PixiEditor.Layer.ReferenceLayerDoesntExistAndHasClipboardContent")]
+    public unsafe void PasteReferenceLayer(DataObject data)
+    {
+        var doc = Owner.DocumentManagerSubViewModel.ActiveDocument;
+        if (doc is null)
+            return;
+
+        var surface = (data == null ? ClipboardController.GetImagesFromClipboard() : ClipboardController.GetImage(data)).First();
```
Comment (Equbuxu) [inline src/PixiEditor/ViewModels/SubViewModels/Main/LayersViewModel.cs:388]: dispose the surface, also the `PasteReferenceLayer` function isn't unsafe

**Review by Equbuxu — CHANGES_REQUESTED** [review]
- Infinite loading when holding down lmb on the opacity slider body and moving the mouse left and right

https://user-images.githubusercontent.com/29402050/213927045-131f4c34-9294-47db-a578-f71bfe60b33d.mp4

- The close button of commands debug window doesn't work
- The layers window makes it seem like you can drag-n-drop a file there, and you can, but only if you put it at the very and, and only if you don't have that many layers already. Would be better if you could insert a file anywhere in the layers list

File: src/PixiEditor/ViewModels/SubViewModels/Main/ClipboardViewModel.cs [hunk 2]
```diff
@@ -26,21 +27,32 @@ public void Cut()
         doc.Operations.DeleteSelectedPixels(true);
     }
 
-    [Command.Basic("PixiEditor.Clipboard.Paste", "Paste", "Paste from clipboard", CanExecute = "PixiEditor.Clipboard.CanPaste", Key = Key.V, Modifiers = ModifierKeys.Control)]
-    public void Paste()
+    [Command.Basic("PixiEditor.Clipboard.Paste", false, "Paste", "Paste from clipboard", CanExecute = "PixiEditor.Clipboard.CanPaste", Key = Key.V, Modifiers = ModifierKeys.Shift)]
+    [Command.Basic("PixiEditor.Clipboard.PasteAsNewLayer", true, "Paste as new layer", "Paste from clipboard as new layer", CanExecute = "PixiEditor.Clipboard.CanPaste", IconPath = "$PixiEditor.Clipboard.Paste", Key = Key.V, Modifiers = ModifierKeys.Control)]
```
Comment (flabbet) [inline src/PixiEditor/ViewModels/SubViewModels/Main/ClipboardViewModel.cs:31]: Paste as new layer doesn't work

File: src/PixiEditor/ViewModels/SubViewModels/Main/DebugViewModel.cs [hunk 3]
```diff
@@ -130,6 +132,21 @@ public void ValidateShortcutMap()
...
+        Owner.FileSubViewModel.RecentlyOpened.Clear();
+        IPreferences.Current.UpdateLocalPreference(PreferencesConstants.RecentlyOpened, Array.Empty<object>());
+    }
+
+    [Command.Debug("PixiEditor.Debug.OpenCommandDebugWindow", "Open command debug window", "Open command debug window")]
+    public void OpenCommandDebugWindow()
+    {
+        Mouse.OverrideCursor = Cursors.Wait;
+        new CommandDebugPopup().Show();
+        Mouse.OverrideCursor = null;
```
Comment (flabbet) [inline src/PixiEditor/ViewModels/SubViewModels/Main/DebugViewModel.cs:147]: xD

File: src/PixiEditor/Views/Dialogs/CommandDebugPopup.xaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+                      ResizeBorderThickness="{x:Static SystemParameters.WindowResizeBorderThickness}" />
+    </WindowChrome.WindowChrome>
+
+    <DockPanel Background="{StaticResource AccentColor}" Focusable="True">
+        <b:Interaction.Behaviors>
+            <behaviours:ClearFocusOnClickBehavior />
+        </b:Interaction.Behaviors>
+
+        <local:DialogTitleBar DockPanel.Dock="Top"
+                              CloseCommand="{x:Static SystemCommands.CloseWindowCommand}"
```
Comment (flabbet) [inline src/PixiEditor/Views/Dialogs/CommandDebugPopup.xaml:28]: Close button doesn't work, probably you forgot to initialize system commands for this window

File: src/PixiEditor/Views/Dialogs/CommandDebugPopup.xaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
+﻿<Window x:Class="PixiEditor.Views.Dialogs.CommandDebugPopup"
```
Comment (flabbet) [inline src/PixiEditor/Views/Dialogs/CommandDebugPopup.xaml:1]: Searchbar of command would be nice too, but it's no must

**Review by flabbet — CHANGES_REQUESTED** [review]
Drag n drop image into layers window works only once per document, if you undo after applying transform to first pasted image

**Review by flabbet — CHANGES_REQUESTED** [review]
1. Open PixiEditor
2. Create new Document
3. Have copied image in clipboard
4. Ctrl + V
5. Don't click apply transform
6. Ctrl + v again

**Comment by flabbet** [comment]
The rest of issues, seems to be fixed

**Review by flabbet — APPROVED** [review]
Yup, seems to be fixed

**Comment by Equbuxu** [comment]
The opacity slider infinite loading problem is still there, but it seems like that's the only issue

**Review by Equbuxu — CHANGES_REQUESTED** [review]
After dragging the opacity slider you can no longer change opacity by clicking the slider body. But hey, at least it doesn't cause infinite loading anymore

**Review by Equbuxu — APPROVED** [review]

**Pull request merged**